typed/markers.go

```diff
@@ -0,0 +1,321 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package typed
+
+import (
+	"fmt"
+	"sync"
+
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/v4/schema"
+	"sigs.k8s.io/structured-merge-diff/v4/value"
+)
+
+const (
+	// markerKey is the key used to store marker values in the object.
+	markerKey = "k8s_io__value"
```

Comment:
@deads2k @sttts Suggestions welcome on how to name the key/value of the markers used in the proposal.  BTH, I'm actively looking for a better string.  Having a key that is also a valid CEL identifier is important for MutatingAdmissionPolicy, so `[a-z][A-Z][0-9]_` (non-number first char) is strongly preferred.